tests/foreman/ui/test_virt_who_config.py

```diff
@@ -205,3 +229,176 @@ def test_positive_lastest_out_of_date(self):
             The 2 virt-who configs are list in the latest out of date section
             of the widget.
         """
+
+
+@run_in_one_thread
+class VirtWhoConfigDeployedTestCase(UITestCase):
+    """Implements Virt-who-config UI tests with virt-who config deployed"""
+
+    @classmethod
+    def set_session_org(cls):
+        """Create global test case organization"""
+        cls.session_org = entities.Organization().create()
+
+    @classmethod
+    @skip_if_not_set('clients', 'fake_manifest')
+    def setUpClass(cls):
+        """setup virt-who virtual machine"""
+        super(VirtWhoConfigDeployedTestCase, cls).setUpClass()
+        cls.lce = entities.LifecycleEnvironment(
+            organization=cls.session_org).create()
+        cls.hypervisor_hostname = settings.compute_resources.libvirt_hostname
+        # Create a new virt-who config
+        virt_who_config = make_virt_who_config({
+            'organization-id': cls.session_org.id,
+            'hypervisor-type': VIRT_WHO_HYPERVISOR_TYPES['libvirt'],
+            'hypervisor-server': 'qemu+ssh://{0}/system'.format(
+                cls.hypervisor_hostname),
+            'hypervisor-username': 'root',
+        })
+        cls.virt_who_vm = VirtualMachine()
+        cls.virt_who_vm.create()
+        # configure virtual machine and setup virt-who service
+        cls.virt_who_data = virt_who_hypervisor_config(
+            virt_who_config['general-information']['id'],
+            cls.virt_who_vm,
+            org_id=cls.session_org.id,
+            lce_id=cls.lce.id,
+            hypervisor_hostname=cls.hypervisor_hostname,
+            configure_ssh=True,
+            subscription_name=VDC_SUBSCRIPTION_NAME,
+            exec_one_shot=True,
+        )
+        cls.virt_who_hypervisor_host = entities.Host(
+            id=cls.virt_who_data['virt_who_hypervisor_host']['id']).read()
+
+    @classmethod
+    def tearDownClass(cls):
+        """destroy the virt-who virtual machine"""
+        cls.virt_who_vm.destroy()
+        super(VirtWhoConfigDeployedTestCase, cls).tearDownClass()
+
+    @tier4
+    def test_positive_search_by_hypervisor(self):
+        """Able to found hypervisor or not hypervisor content host by searching
+        hypervisor = true or hypervisor = false
+
+        :id: 2165023f-7184-400c-a9d9-e0c0d065a7d2
+
+        :expectedresults: the hypervisor search work properly
+
+        :BZ: 1246554, 1495271
+
+        :CaseLevel: System
+        """
+        with Session(self) as session:
+            set_context(session, org=self.session_org.name)
+            is_hypervisor_hosts = {
+                'true': self.virt_who_hypervisor_host.name,
```

Comment:
why do you use string keys and then use that ugly if condition? I think we can use boolean ones

Reply:
and `str(not is_hypervisor_hosts).lower()` correspondingly